pbeApp/pbexport.cpp

```diff
@@ -362,42 +442,66 @@ void PBWriter::prepFile()
 
     escapingarraystream encbuf;
     {
-        google::protobuf::io::CodedOutputStream encstrm(&encbuf);
-        header.SerializeToCodedStream(&encstrm);
+    	if (!fileexists) {
+			google::protobuf::io::CodedOutputStream encstrm(&encbuf);
+			header.SerializeToCodedStream(&encstrm);
+    	}
     }
     encbuf.finalize();
 
-    outpb.open(fname.str().c_str());
-    outpb.write(&encbuf.outbuf[0], encbuf.outbuf.size());
+    outpb.open(fname.str().c_str(), std::fstream::app);
+    if (!fileexists) { //if file exists do not write header
+    	outpb.write(&encbuf.outbuf[0], encbuf.outbuf.size());
+    }
 }
 
-PBWriter::PBWriter(DataReader& reader)
+PBWriter::PBWriter(DataReader& reader, stdString pv)
     :reader(reader)
 	,info(reader.getInfo())
 	,year(0)
+    ,name(pv)
 {
     samp = reader.get();
 }
 
 void PBWriter::write()
 {
+	typeChangeError = 0;
     while(samp) {
-        prepFile();
-        (*transcode)(*this);
-        bool ok = outpb.good();
-        outpb.close();
-        if(!ok) {
-            std::cerr<<"Error writing file\n";
-            break;
-        }
+    	try {
+			prepFile();
+			if (!samp) break;
+			(*transcode)(*this);
+    	} catch (std::exception& up) {
+    		if (std::strstr(up.what(),"Error in data header")) {
```

Comment:
It's generally not a good idea to depend on the text of exceptions.  It's better to catch by type if possible.  'typeid(up).name()' is helpful if you don't know the type of the exception.